src/components/giftcertificateform.main.jsx

```diff
@@ -0,0 +1,291 @@
+              Authorization: localStorage.getItem(`${Config.cortexApi.scope}_oAuthToken`),
+            },
+            body: JSON.stringify({
+              'gift-certificate-code': el,
+            }),
+          })
+          .then(res => res.json())
+          .then((res) => {
+            if (res.code) {
+              const data = { ...res, isChecked: false };
+              this.setState(prevState => ({
+                giftCertificateEntity: [...prevState.giftCertificateEntity, data],
+                giftCertificatesCodeArr: [...prevState.giftCertificatesCodeArr, data.code],
+              }));
+            }
+          })
+          .catch((error) => {
+            // eslint-disable-next-line no-console
+            console.error(error.message);
+          });
+      });
+    });
+  }
+
+  setGiftCertificatesCode(event) {
+    this.setState({ giftCertificatesCode: event.target.value, showErrorMsg: false });
+  }
+
+  getGiftCertificateEntity() {
+    const { giftCertificatesCode, giftCertificateEntity } = this.state;
+    if (giftCertificateEntity.filter(el => el.code === giftCertificatesCode).length === 0) {
+      this.setState({ showLoader: true });
+      login().then(() => {
+        cortexFetch(`/giftcertificates/${Config.cortexApi.scope}/lookup/form?followlocation=true`,
+          {
+            method: 'post',
+            headers: {
+              'Content-Type': 'application/json',
+              Authorization: localStorage.getItem(`${Config.cortexApi.scope}_oAuthToken`),
+            },
+            body: JSON.stringify({
+              'gift-certificate-code': giftCertificatesCode,
+            }),
+          })
+          .then(res => res.json())
+          .then((res) => {
+            if (res.code) {
+              const data = { ...res, isChecked: false };
+              this.setState(prevState => ({
+                open: false,
+                giftCertificatesCode: '',
+                giftCertificateEntity: [...prevState.giftCertificateEntity, data],
+                showErrorMsg: false,
+                giftCertificatesCodeArr: [...prevState.giftCertificatesCodeArr, data.code],
+                showLoader: false,
+              }));
+              const { giftCertificatesCodeArr } = this.state;
+              localStorage.setItem('giftCertificatesCodeArr', JSON.stringify(giftCertificatesCodeArr));
+            } else {
+              this.setState({ showErrorMsg: true, showLoader: false });
+            }
+          })
+          .catch((error) => {
+            // eslint-disable-next-line no-console
+            console.error(error.message);
+          });
+      });
+    } else {
+      this.setState({ showErrorMsg: true });
+    }
+  }
+
+  handleCloseModal() {
+    this.setState({ open: false });
+  }
+
+  handleOpenModal() {
+    this.setState({ open: true });
+  }
+
+  handleCheck(el, index) {
+    const { giftCertificateEntity, chosenGiftCertificates } = this.state;
```

Comment:
After this is checked, does it actually add the cert to the order? I'm not seeing it reflected. Still looking through the code further though.